src/routes/safe/components/Transactions/TxList/TxSummary.tsx

```diff
@@ -10,21 +10,77 @@ import {
   isMultiSendTxInfo,
   isMultiSigExecutionDetails,
 } from 'src/logic/safe/store/models/types/gateway.d'
-import { InlineEthHashInfo } from './styled'
+import { InlineEthHashInfo, StyledGridRow } from './styled'
 import { NOT_AVAILABLE } from './utils'
 import TxShareButton from './TxShareButton'
 import { IS_PRODUCTION } from 'src/utils/constants'
 import TxInfoMultiSend from './TxInfoMultiSend'
 import DelegateCallWarning from './DelegateCallWarning'
+import styled from 'styled-components'
+import { getByteLength } from 'src/components/DecodeTxs' // must pass to utils
+import { generateSignaturesFromTxConfirmations } from 'src/logic/safe/safeTxSigner'
+import { makeConfirmation } from 'src/logic/safe/store/models/confirmation'
 
 type Props = { txDetails: ExpandedTxDetails }
 
+const FlexWrapper = styled.div<{ margin: number }>`
+  display: flex;
+  align-items: center;
+
+  > :nth-child(2) {
+    margin-left: ${({ margin }) => margin}px;
+  }
+`
+
+type TxDataRowType = { inlineType?: 'hash' | 'rawData'; title: string; value: string }
+
+const TxDataRow = ({ inlineType, title, value }: TxDataRowType) => (
+  <StyledGridRow>
+    <Text size="xl" as="span">
+      {title}
+    </Text>
+    {inlineType === 'hash' && (
+      <InlineEthHashInfo textSize="xl" hash={value} shortenHash={8} showCopyBtn explorerUrl={getExplorerInfo(value)} />
+    )}
+    {inlineType === 'rawData' && (
+      <FlexWrapper margin={5}>
+        <Text size="xl">{value ? getByteLength(value) : 0} bytes</Text>
+        <CopyToClipboardBtn textToCopy={value || ''} />
```

Comment:
I'd suggest conditionally adding this instead of allowing an empty string to be copied, e.g.

```ts
{value && <CopyToClipboardBtn textToCopy={value} />}
```